modules/user_accounts/php/edit_user.class.inc

```diff
@@ -810,31 +810,38 @@ class Edit_User extends \NDB_Form
     }
 
     /**
-     * Query database on UserID to check that password has doesn't
-     * exist and account is still pending approval, to ensure the
-     * users with account activity can't be rejected by setting
-     * their status to pending.
+     * Query database on UserID to check that password hash does not
+     * exist and account is still pending approval.
      *
-     * @param string $id identifier of account to verify
+     * This ensures that users with account activity cannot be rejected
+     * if their approval status is, for some reason, set to pending.
      *
-     * @return boolean true is Password_hash is null and
-     * Pending_approval is Yes, false otherwsie
+     * @param string $userID of account to verify
+     *
+     * @return boolean true if 'Password_hash' is null and
+     *         'Pending_approval' is Yes. False, otherwise.
+     *
+     * @throws Loris exception if the userID is not found in the database.
      */
-    function _canRejectAccount($id)
+    static function canRejectAccount($userID)
     {
-        $DB     = \Database::singleton();
-        $fields =  $DB->pselect(
-            "SELECT Password_hash, Pending_approval
-                   FROM users
-                   WHERE UserID=:id",
-            array("id" => $id)
-        );
-        if (is_null($fields[0]['Password_hash'])
-            && $fields[0]['Pending_approval']=="Y"
-        ) {
-            return true;
+        $DB = \Database::singleton();
+        try{
+            $fields =  $DB->pselectRow(
+                "SELECT Password_hash, Pending_approval
+                           FROM users
+                           WHERE UserID=:id",
+                array("id" => $userID)
+            );
+            if (empty($fields['Password_hash'])
+                && $fields['Pending_approval']=="Y"
+            ) {
+                return true;
+            }
+            return false;
+        } catch (DatabaseException $e) {
```

Comment:
@zaliqarosli when is this catch ever reached ? I thought that when a select has no results it only returns empty, not an error. any usecases for it ?

Reply:
@ridz1208 i guess it doesn't ever. I'll look into it. Here's another question: when do we
`$DB = \Database::singleton();` and when do we
`$DB_factory = \NDB_Factory::singleton();
$DB = $DB_factory->database();`??

Reply:
@ridz1208 resolved. please re-review :)

Reply:
@ridz1208 on a second look, the error is for when the $userID does not exist in the database at all, not when the select returns empty

Reply:
to my knowledge none of the statements in the `try{}catch` throw an exception... if $userid does not exist, the pselectRow returns empty thats it

@driusan am I missing something ?

Reply:
@ridz1208 gotcha.. pselectRow doesn't throw a DatabaseException but maybe it should?

Reply:
sure... u should add it... 

*He says as he types up his resignation letter*

Reply:
No, it should only throw exceptions in exceptional circumstances (ie. invalid SQL), an empty data set is a normal thing for a query to return. (It should probably return null and not an empty array, but this isn't the place to change that..)

I agree with @ridz1208 that I can't see any way for the catch to occur. If the underlaying implementation threw an exception (ie. if the PDO connection died while running it), it wouldn't be a DatabaseException.